backend/learnings/views.py

```diff
@@ -45,3 +48,36 @@ def create(self, request):
         except Exception:
             content = {"message": "Account Not Registered"}
             return Response(content, status=status.HTTP_400_BAD_REQUEST)
+
+
+class DisplayCategeroyWithUserFilter(APIView):
+    permission_classes = [IsAuthenticated]
+    authentication_classes = [TokenAuthentication]
+
+    def get(self, request, user):
+        categories_to_display = []
+        taken = []
+        categories = []
+
+        user = EUser.objects.get(username=user)
+        taken_quizzes = QuizTaken.objects.filter(user_id=user.id)
+        all_categories = Category.objects.all()
+
+        for quiz in taken_quizzes:
+            taken.append(quiz.id)
+
+        for category in all_categories:
+            categories.append(category.id)
+
+        category_ids_not_taken = set(taken) ^ set(categories)
+
+        for id in category_ids_not_taken:
+            get_category = Category.objects.get(id=id)
+            categories_to_display.append(
+                {
+                    "id": get_category.id,
+                    "category_name": get_category.category_name,
+                    "description": get_category.description,
+                }
+            )
```

Comment:
I think we can write this in one query, try looking up `django queries`
And Im pretty sure line 72 is a bug, one is `taken = quiz_ids` and `categories = category_ids`